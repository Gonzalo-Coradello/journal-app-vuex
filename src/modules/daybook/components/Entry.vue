<template>
    <div class="entry-container mb-3 pointer p-2" @click="$router.push({ name: 'entry', params: { id: 10 } })">
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">28</span>
            <span class="mx-1 fs-5">Diciembre</span>
            <span class="mx-2 fw-light">2023, jueves</span>
        </div>

        <div class="entry-description">
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Vitae explicabo quo, perspiciatis, culpa excepturi iste ab mollitia eligendi reprehenderit ut veritatis sed rerum repellendus hic temporibus earum doloremque unde recusandae voluptatibus dolorem repudiandae sint. Rerum quaerat praesentium doloremque, voluptates ex deserunt earum architecto voluptatibus illum harum nihil reprehenderit minus non.
        </div>
    </div>
</template>

<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2 all ease-in;

    &:hover {
        background-color: lighten($color: grey, $amount: 45);
    }

    .entry-description {
        font-size: 12px;
    }
}
</style>