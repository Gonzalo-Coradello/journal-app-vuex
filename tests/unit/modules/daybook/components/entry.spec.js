import { beforeEach, describe, expect, test, vi } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from '../../../mock-data/test-journal-state'

describe('Pruebas en Entry Component', () => {
  let wrapper

  const mockRouter = {
    push: vi.fn()
  }

  beforeEach(() => {
    wrapper = shallowMount(Entry, {
      props: {
        entry: journalState.entries[0]
      },
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })
  })

  test('debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe redireccionar al hacer click en .entry-container', () => {
    wrapper.find('.entry-container').trigger('click')

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: { id: journalState.entries[0].id }
    })
  })

  test('pruebas en las propiedades computadas', () => {
    const { day, month, yearDay } = wrapper.vm

    expect(day).toBe(4)
    expect(month).toBe('Enero')
    expect(yearDay).toBe('2024, Jueves')
  })
})
