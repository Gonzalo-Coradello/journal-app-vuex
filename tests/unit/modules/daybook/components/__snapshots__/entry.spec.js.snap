// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en Entry Component > debe hacer match con el snapshot 1`] = `
"<div data-v-4a9e2957="" class="entry-container mb-3 pointer p-2">
  <div data-v-4a9e2957="" class="entry-title d-flex"><span data-v-4a9e2957="" class="text-success fs-5 fw-bold">4</span><span data-v-4a9e2957="" class="mx-1 fs-5">Enero</span><span data-v-4a9e2957="" class="mx-2 fw-light">2024, Jueves</span></div>
  <div data-v-4a9e2957="" class="entry-description">Hoy es un buen d√≠a para aprender Vue</div>
</div>"
`;
